samtranslator/model/api/http_api_generator.py

```diff
@@ -435,6 +435,20 @@ def _construct_record_sets_for_domain(
 
         return recordset_list
 
+    def _construct_route53_routing_policy_properties(
+        self, route53_config: Dict[str, Any], recordset: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        if route53_config.get("Region") is not None:
+            if route53_config.get("SetIdentifier") is None:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "SetIdentifier is required in order to enable latency based routing on Route53 recordsets.",
+                )
```

Comment:
Is this test necessary?

I have no strong preference either way (it's often fine to let the underlying resource figure out if it's wrong), but if we do add this, there are 2 minor caveats:

1. We should add a transform test for this error case (see https://github.com/aws/serverless-application-model/pull/2885 for an example, basically create a transform test template prefixed with `error_`, and in the output create a JSON with a top-level `errorMessage` with the expected error message)
2. If in the future `Region` becomes usable without `SetIdentifier`, it'll be a poor experience until we change the logic (releases can take some time)

Reply:
As CloudFormation performs equivalent validation after the template is transformed upon second thought I don't think this test is necessary; especially given your second point. Thanks!